modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/rule/logical/ExposeIndexRule.java

```diff
@@ -79,6 +83,14 @@ public void onMatch(RelOptRuleCall call) {
         call.transformTo(indexes.get(0), equivMap);
     }
 
+    /** Filter pre known not applicable variants. Significant shrink search space in some cases. */
+    private static boolean filter(IgniteTable table, String idxName, List<SearchBounds> searchBounds) {
```

Comment:
With such approach pk index won't be chosen for CNLJ

Reply:
seems I was wrong about pk index would not being chosen for CNLJ... Please ignore message above...